content/src/controller.js

```diff
@@ -1394,7 +1420,7 @@ function doneWithFile(filename) {
     }
   } else {
     if (filename.indexOf('/') >= 0) {
-      filename = filename.replace(/\/[^\/]+\/?$/, '');
+      filename = filename.replace(/[^\/]+\/?$/, '');
```

Comment:
The reason behind this is that `history.state.previous` is something like `/edit/` or `/edit/folder/`, but we were stripping the `/` at the end, thus the comparison below fails, so the unsaved alert is not triggered. 

So the fix is to just keep the `/`.